File: backend/models/crypto.go
package models

// CryptoData represents the structure of cryptocurrency data
type CryptoData struct {
	ID                       string  `json:"id"`
	Symbol                   string  `json:"symbol"`
	Name                     string  `json:"name"`
	Image                    string  `json:"image"`
	CurrentPrice             float64 `json:"current_price"`
	MarketCap                float64 `json:"market_cap"`
	PriceChangePercentage24H float64 `json:"price_change_percentage_24h"`
}

File: backend/models/crypto.js
import mongoose from "mongoose";

const cryptoSchema = new mongoose.Schema({
  id: { type: String, required: true },
  symbol: { type: String, required: true },
  name: { type: String, required: true },
  image: { type: String, required: true },
  current_price: { type: Number, required: true },
  market_cap: { type: Number, required: true },
  price_change_percentage_24h: { type: Number, required: true },
});

const Crypto = mongoose.model("Crypto", cryptoSchema);

export default Crypto;
